onsstove/onsstove.py

```diff
@@ -410,13 +410,56 @@ def read_scenario_data(self, path_to_config: str, delimiter=','):
                         config[row['Param']] = float(row['Value'])
                     elif row['data_type'] == 'string':
                         config[row['Param']] = str(row['Value'])
+                    elif row['data_type'] == 'bool':
+                        config[row['Param']] = str(row['Value']).lower() in ['true', 't', 'yes', 'y', '1']
                     else:
                         raise ValueError("Config file data type not recognised.")
         if self.specs is None:
             self.specs = config
         else:
             self.specs.update(config)
 
+    def set_base_fuel(self, techs: list = None):
+        """
+        Defines the base fuel properties according to the technologies
+        tagged as is_base = True or a list of technologies as input.
+        If no technologies are passed as input and no technologies are tagged
+        as is_base = True, then it calculates the base fuel properties considering
+        all technologies in the model
+        """
+        if techs is None:
+            techs = self.techs.values()
+        base_fuels = []
+        for tech in techs:
+            if tech.is_base:
+                base_fuels.append(tech)
+        if len(base_fuels) == 1:
+            self.base_fuel = base_fuels[0]
+        else:
+            if len(base_fuels) == 0:
+                base_fuels = techs
+            base_fuel = Technology(name='Base fuel')
+            base_fuel.carbon = 0
+            base_fuel.total_time_yr = 0
+            for tech in base_fuels:
+                if tech.carbon is None:
+                    tech.carb(self)
+                if tech.total_time_yr is None:
+                    tech.total_time(self)
+                tech.health_parameters(self)
+                current_share = (self.gdf['IsUrban'] > 20) * tech.current_share_urban
+                current_share[self.gdf['IsUrban'] < 20] = tech.current_share_rural
+                # for prop in ['carbon', 'pm25', 'total_time_yr']:
+                base_fuel.carbon += tech.carbon * current_share
+                base_fuel.total_time_yr += tech.total_time_yr * current_share
+                for paf in ['paf_alri_r', 'paf_copd_r', 'paf_ihd_r',
```

Comment:
What happens here?

Reply:
I mean can PAF really be multiplied by the shares?

Reply:
Yes this is something I am not sure. I first tried to get the weighted average of the pm25 and then calculate the rr and paf based on that, but I think that is not correct, because we would be kind of creating a new fuel that does not exist. So if we instead calculate the paf for each fuel and then multiply by the shares, we would be accounting for the population attributable factors of each fuels and then scaling this values to their population share only.